app/main.py

```diff
@@ -1 +1,38 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, herbivore: Herbivore) -> None:
+        if (
+                not isinstance(herbivore, Carnivore)
+                and not herbivore.hidden
+                and self.health > 0
+        ):
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                herbivore.health = 0
+                Animal.alive.remove(herbivore)
+                del herbivore
```

Comment:
dead animal doesn't need health attr
```suggestion
            if herbivore.health <= 0:
                Animal.alive.remove(herbivore)
                del herbivore
```